services/provider/client/client_mock.go

```diff
@@ -0,0 +1,87 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/red-hat-storage/ocs-operator/services/provider/common"
+	pb "github.com/red-hat-storage/ocs-operator/services/provider/pb"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type MockOCSProviderClient struct {
```

Comment:
Why do we need a MOCK provider client?

Reply:
Actually, we are creating this for testing purposes as our API server and controller work is blocked for now.